pkg/agent/util.go

```diff
@@ -22,11 +25,22 @@ func getTimestamp() v1Time.Time {
 	return newV1Time
 }
 
-// ApplyResouceToConfig - applies the resources to agent configs
+// ApplyResourceToConfig - applies the resources to agent configs
 // Uses reflection to get the IResourceConfigCallback interface on the config struct or
 // struct variable.
 // Makes call to ApplyResources method with dataplane and agent resources from API server
-func ApplyResouceToConfig(cfg interface{}) error {
+func ApplyResourceToConfig(cfg interface{}) error {
+	obj := cfg
+	defer func() {
+		if err := recover(); err != nil {
+			str := fmt.Sprintf("%v", err)
+			if strings.Contains(str, "nil pointer dereference") {
+				log.Errorf("The function 'ApplyResources' for interface IResourceConfigCallback is not implemented in %s.", reflect.TypeOf(obj))
+			}
+		}
+		// }
```

Comment:
Comment

Reply:
Good. I meant to do that, as this was kind of crazy. Thx.